mediaire_toolbox/transaction_db/transaction_db.py

```diff
@@ -125,28 +125,55 @@ def set_processing(self,
             self.session.rollback()
             raise
 
-    def set_failed(self, id_: int, cause: str):
+    def set_failed(self, id_: int, cause: str = None):
         """to be called when a transaction fails. Save error information
         from 'cause'"""
         try:
             t = self._get_transaction_or_raise_exception(id_)
             t.task_state = TaskState.failed
             t.end_date = datetime.datetime.utcnow()
-            t.error = cause
+            if cause:
+                t.error = cause
             self.session.commit()
         except:
             self.session.rollback()
             raise
 
-    def set_completed(self, id_: int):
+    def set_completed(self, id_: int, clear_error: bool = True):
         """to be called when the transaction completes successfully.
         Error field will be set explicitly to '' and end_date automatically
         adjusted."""
         try:
             t = self._get_transaction_or_raise_exception(id_)
             t.task_state = TaskState.completed
             t.end_date = datetime.datetime.utcnow()
-            t.error = ''
+            if clear_error:
+                t.error = ''
+            self.session.commit()
+        except:
+            self.session.rollback()
+            raise
+
+    def set_skipped(self, id_: int, cause: str = None):
+        """to be called when the transaction is skipped.
```

Comment:
I'm confused here. the doc says error field will be set to '' but we actually have an skipped cause error as parameter?